src/libs/conduit/conduit_node.cpp

```diff
@@ -13501,6 +13502,146 @@ Node::find_first_data_ptr() const
+                    res = false;
+                }
+            }
+        }
+
+        child_itr = n.children();
+        while(child_itr.has_next())
+        {
+            const conduit::Node &n_child = child_itr.next();
+            const std::string child_path = child_itr.name();
+
+            Node &info_child = info[child_path];
+            if(!has_child(child_path))
+            {
+                info_child.set("Missing from self.");
+                res = false;
+            }
+            else
+            {
+                if(fetch(child_path).compare(n_child, info_child))
+                {
+                    info.remove(child_path);
+                }
+                else
+                {
+                    res = false;
+                }
+            }
+        }
+    }
+    else if(m_dtid == DataType::LIST_ID)
+    {
+        // NOTE: This case is a bit strange in the comparison; fortunately, it
+        // can be distilled to two primary cases:
+        // - Nodes Have 0 Diffs: The info node is completely empty.
+        // - Nodes Have >1 Diffs: The info node is a list where the differing
+        //   nodes between the comparison parents are filled in the list.
+        //   All other nodes in the list are empty.
+
+        index_t m_nchild = number_of_children();
+        index_t n_nchild = n.number_of_children();
+
+        size_t i = 0;
+        for(; i < (size_t)std::min(m_nchild, n_nchild); i++)
+        {
+            const Node &m_child = child(i);
+            const Node &n_child = n.child(i);
+            Node &info_child = info.append();
+            res &= m_child.compare(n_child, info_child);
+        }
+
+        res &= m_nchild != n_nchild;
+        for(; i < (size_t)std::max(m_nchild, n_nchild); i++)
+        {
+            Node &info_child = info.append();
+            std::string loc_str = (i >= (size_t)m_nchild) ? "self" : "other";
+            info_child.set("Missing from " + loc_str);
+        }
+    }
+    else if(!dtype().compatible(n.dtype()) || !n.dtype().compatible(dtype()))
+    {
+        info.set("Data type incompatibility.");
+        res = false;
+    }
+    else
+    {
+        // TODO(JRC): Determine how to customize this function so that users that
+        // want the stride/endianness to be the same can still perform a proper query.
+        DataType elem_dtype(dtype().id(), 1, 0, 0, dtype().element_bytes(), dtype().endianness());
+        for(size_t i = 0; i < (size_t)dtype().number_of_elements(); i++)
+        {
+            const conduit::Node m_elem(elem_dtype, (void*)element_ptr(i), true);
+            const conduit::Node n_elem(elem_dtype, (void*)n.element_ptr(i), true);
+            // TODO(JRC): Figure out how to make this work to enable better
+            // element-by-element comparisons.
+            if(m_elem.value() != n_elem.value())
+            {
+                info.set("Data value mismatch.");
+                res = false;
+            }
+        }
```

Comment:
this is the hardest key to the puzzle, `m_elem.value() != n_elem.value() `won't work b/c each `.value()` requires a constraining type to work. 

One solution I thought  for these type of general problems was to have a `void_array`, but for various reasons I am evolving to think this is not a good idea. (https://github.com/LLNL/conduit/issues/98)